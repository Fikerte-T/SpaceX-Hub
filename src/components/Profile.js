import React from 'react';
import RocketProfile from './RocketProfile';
import MissionsProfile from './MissionsProfile';

const Profile = () => (
  <div className="profile-container">
    <MissionsProfile />
    <RocketProfile />
  </div>
);

export default Profile;
