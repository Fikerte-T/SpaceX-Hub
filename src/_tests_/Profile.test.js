import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/Profile';

describe('Mission Element Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(screen.queryByText(/My Rockets/)).toBeTruthy();
    expect(screen.queryByText(/My Missions/)).toBeTruthy();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
