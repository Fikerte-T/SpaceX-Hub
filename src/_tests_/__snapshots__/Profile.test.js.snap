// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Element Tests renders correctly 1`] = `
<div
  className="profile-container"
>
  <div>
    <h2>
      My Missions
    </h2>
  </div>
  <div>
    <h2>
      My Rockets
    </h2>
  </div>
</div>
`;
